### DISCUSSION TITLE
X11 Clipboard Forwarding

### DISCUSSION THREAD

**Discussion by FenycFur** [wezterm/wezterm#5042] [Q&A]
### What Operating System(s) are you running on?

Windows, Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

On Windows 10:
VcXsrv X Server v 1.20.14.0 (31 Dec 2021)

On Linux:
```
$ xdpyinfo | grep version
version number:    11.0
```

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Does WezTerm support setting the $DISPLAY variable and forwarding an X11 server?
I'd like to have clipboard shared between Windows guest and Linux host. I haven't found anything in documentation or forums to suggest this exists.
Or if it's not supported, what's the correct/standard way of syncing clipboards between Linux host and Windows guest?

**Comment by wez** [comment]
The linux version of wezterm supports X11 and respects the $DISPLAY variable.
It also supports Wayland, and that will be used by default in preference to X11.
You can disable Wayland support via https://wezfurlong.org/wezterm/config/lua/config/enable_wayland.html
The clipboard is owned by whichever window system is in use; either Wayland OR X11.

The linux version has no knowledge of Windows, so sharing clipboard between X11 and Windows is purely a function of whatever environment you may be running in; it is outside the scope of what wezterm is aware of.

**Reply by FenycFur** [reply]
So when connecting from Linux host to Linux host, x11 forwarding should work. but not when connecting from Windows host to Linux host?
